element/src/commonMain/kotlin/Element.kt

```diff
@@ -0,0 +1,67 @@
+package com.juul.krayon.element
+
+import com.juul.krayon.kanvas.Kanvas
+
+public abstract class Element {
```

Comment:
So an `Element` is essentially a node on a tree, and it knows where and how to draw itself?

Reply:
Yup. D3 primarily operates on the DOM, which we don't have, so this is an attempt at creating a vector-specific, platform-agnostic analogue. I haven't filled out all the different element types, but generally there should be one for each "operation" in `Kanvas`, like `drawRect` -> `RectangleElement` or `pushTransform` -> `TransformElement`. These are themselves mostly similar to SVG's `<rect>` and `<g>`, although it won't be exactly 1:1 for every element.